### PR TITLE
in-tree IRLock driver

### PR DESCRIPTION
**Pull request by tridge**
This creates an in-tree IRLock driver to replace the PX4 one
Note tested yet, waiting for tests by Randy

### CODE REVIEW

File: libraries/AP_IRLock/AP_IRLock_I2C.cpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+extern const AP_HAL::HAL& hal;
+
+#define IRLOCK_I2C_ADDRESS		0x54
+
+#define IRLOCK_SYNC			0xAA55AA55
+
+void AP_IRLock_I2C::init()
+{
+    dev = std::move(hal.i2c_mgr->get_device(1, IRLOCK_I2C_ADDRESS));
```
Comment (OXINARF) [inline libraries/AP_IRLock/AP_IRLock_I2C.cpp]: Why only look at bus 1? Actually weren't this looking at bus 0 before (and eventually 1 for VR Brain boards)?

Comment (tridge) [inline libraries/AP_IRLock/AP_IRLock_I2C.cpp]: it was only looking at the external bus (bus 0 with the new bus numbers) before.
We can certainly extend it to bus 0, but the problem with this sensor is it has no way to properly detect the sensor. There is no WHOAMI or similar. So you can't just probe buses. We'd need a parameter to say which bus its on.

**Comment by tridge** [comment]
randy had a good flight with this today, getting 3 out of 3 good precision landings

File: libraries/AP_IRLock/AP_IRLock_I2C.cpp (hunk 1 above)
Comment (OXINARF) [inline libraries/AP_IRLock/AP_IRLock_I2C.cpp]: I think we should add a parameter then. I think that we shouldn't make a driver that relies on a board only having one external I2C bus.

Comment (tridge) [inline libraries/AP_IRLock/AP_IRLock_I2C.cpp]: I agree. IRLock currently doesn't have any parameters at all, and it being tied to one bus isn't the only problem with the driver. 
The aim of this PR is to get the driver in-tree with existing functionality, then we can improve on it. It should be converted to use a frontend-backend split like other drivers, and get parameters (at least for bus address).
The way it handles health status also needs fixing. The old driver always reported as healthy. I changed that to report healthy when it had received a target in the last 0.1s. That works, but MissionPlanner saw reports of errors from precision land when it went unhealthy (when it couldn't see a target).

**Comment by OXINARF** [comment]
Merged together with #5255

**Comment by tridge** [comment]
my apologies, I accidentally pushed an earlier version of this PR as part of another push. I will push the fixes to take it to the level that Randy successfully test flew.

**Comment by OXINARF** [comment]
Oh, I though it was the same. I'll open this just as reminder.

**Comment by tridge** [comment]
now its in master properly :-)

**Comment by rmackay9** [comment]
By the way, I flight tested this and it worked really well.

**Pull request closed without merging**